pkg/plugin/trans.go

```diff
@@ -214,6 +214,33 @@ func (r resourceTransition[T]) handleDeleted(currentlyMigrating bool) (verdict s
 	return verdict
 }
 
+// handleAutoscalingDisabled updates r.node with changes to clear any buffer and capacityPressure
+// from r.pod
+//
+// A pretty-formatted summary of the changes is returned as the verdict, for logging.
+func (r resourceTransition[T]) handleAutoscalingDisabled() (verdict string) {
+	// buffer is included in reserved, so we reduce everything by buffer.
+	buffer := r.pod.buffer
+	valuesToReduce := []*T{&r.node.reserved, &r.node.buffer, &r.pod.reserved, &r.pod.buffer}
+	for _, v := range valuesToReduce {
+		*v -= buffer
+	}
+
+	r.node.capacityPressure -= r.pod.capacityPressure
+	r.pod.capacityPressure = 0
+
+	fmtString := "pod had buffer %d, capacityPressure %d; " +
+		"node reserved %d -> %d, capacityPressure %d -> %d"
+	verdict = fmt.Sprintf(
```

Comment:
I'd avoid the concat in the fmtstring at least, and probably also avoid not just returning the result of the `fmt.Sprintf` directly.

Reply:
Leaving as-is, to match the rest of the file. May change in a follow-up PR, but undecided.